### ISSUE TITLE
Errors reading SwissFEL JF16M NeXus data

### ISSUE THREAD

**Issue by phyy-nx** [cctbx/dxtbx#537]
I ran into errors today reading SwissFEL JF16M detector data in NeXus format.  See for example the [SwissFEL 68 image dataset](https://doi.org/10.5281/zenodo.3352357) (part of the [gold standard paper](https://doi.org/10.1107/S2052252520008672)).  The JF16M is untested (#93), something that's been on my plate to fix for a while now, but it did used to work, so I tracked down what the problem is.

It looks like #455 made any NeXus file that conforms to NXmx be read by FormatNXmx.py instead of FormatNexus.py.  The intent I think behind #455 was to only have that be the case for Eiger data, but the change to the [understand](https://github.com/cctbx/dxtbx/blob/main/src/dxtbx/format/FormatNXmx.py#L22) method of FormatNXmx meant it's now picked up by any NXmx file, including SwissFEL NeXus files.  It was the intent for FormatNXmx to replace FormatNexus, so it's not a worry.

There are two easily fixable bugs in FormatNXmx (PR incoming):
- The Zenodo JF16M data has a pixel mask that is returned by h5py as a non-contiguous numpy array, which flumpy fails on.  
- Our wrapper code [creates an array of length 1](https://github.com/cctbx/cctbx_project/blob/master/xfel/swissfel/jf16m_cxigeom2nexus.py#L319) for incident_wavelength, which causes FormatNXmx to fail.  FormatNexus handles this in a subtle way, using [`float`](https://github.com/cctbx/dxtbx/blob/main/src/dxtbx/format/nexus.py#L600-L602) as a way to convert an array of shape `(1,)` to a single float value.

Harder to fix, and a big one for diffBragg/nanoBragg, is that FormatNXmx doesn't support per-image spectra.  There's [a lot of code in FormatNexus](https://github.com/cctbx/dxtbx/blob/main/src/dxtbx/format/nexus.py#L560-L635) to handle these spectra, including using variants, and that needs to be ported into FormatNXmx. I have an implementation I'll add to the upcoming pull request that is mostly a copy from FormatNexus.py.  I can provide a dataset to test spectra on, but all I have now is a 92 GB dataset, so I need to trim it down.  I need to do that anyway if I'm going to ever fix #93, so maybe it's time to figure out how to trim an h5 file.

**Comment by phyy-nx** [comment]
I found another problem between FormatNexus and FormatNXmx.  The imported.expt files from the same dataset have different origins for the panels.

How to reproduce:
- Download the [4 image JF16M Zenodo dataset](https://doi.org/10.5281/zenodo.7005251). This is new as of today, and PR dials/data#389 will add it to dials.data.
- Edit FormatNXmx understand to return False
- `dials.import lyso009a_0087.JF07T32V01_master_4img.h5`
- `dials.show imported.expt > a`
- Revert edit and checkout dxtbx branch nxmx_spectra. This only fixes the two easy bugs above and adds spectrum support to the beam, so it shouldn't affect the detector model reading.
- `dials.import lyso009a_0087.JF07T32V01_master_4img.h5`
- `dials.show imported.expt > b`
- `diff a b`

Result: each panel differs by a pixel (75 microns)
```
25c25
<   origin: {-151.524,-164.784,-97.536}
---
>   origin: {-151.599,-164.784,-97.536}
```

**Comment by phyy-nx** [comment]
The off-by-one error was in FormatNexux.py, not in FormatNXmx.  Fixed in 63d5045c62340e9fe0e868cae19d0062a73f6986 as part of #538.
